src/app/main/components/movies-view/movies-view.component.ts

```diff
@@ -44,4 +43,8 @@ export class MoviesViewComponent implements OnInit {
         this.paginationService.setPages(result.page);
       });
   }
+
+  public ngOnDestroy(): void {
+    this.desctroy.unsubscribe();
```

Comment:
You need to emit a value before unsubscribing
`this.destroy$.next()
this.destroy$.unsubscribe()`
I also recommend using type Subject<void>